src/screens/Signup/index.tsx

```diff
@@ -13,24 +17,26 @@ import {
 import logo from './assets/wolox-logo.png';
 import styles from './styles.module.scss';
 
-interface SignupForm {
-  firstName: string;
-  lastName: string;
-  email: string;
-  password: string;
-  passwordConfirmation: string;
-}
-
 function Signup() {
   const { t, i18n } = useTranslation();
+
   const {
     register,
     handleSubmit,
     watch,
     formState: { errors }
-  } = useForm<SignupForm>();
+  } = useForm<User>();
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [state, loading, error, sendRequest] = useLazyRequest({ request: signUp });
+  const errorMessage =
+    error?.problem === 'NETWORK_ERROR'
+      ? 'Hubo un problema al cargar los datos, volvé a intentarlo en unos minutos'
+      : 'Ese email ya está registrado';
```

Comment:
This logic is bound to be repeated in many requests, the api has a monitor to check for this specific case, perhaps it'll be useful.